Samples/excel-keyboard-shortcuts/src/taskpane/taskpane.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+ * See LICENSE in the project root for license information.
+ */
+
+// images references in the manifest
+import "../../assets/icon-16.png";
+import "../../assets/icon-32.png";
+import "../../assets/icon-80.png";
+
+/* global console, document, Excel, Office */
+
+Office.actions.associate("SHOWTASKPANE", function() {
+  return Office.addin
+    .showAsTaskpane()
+    .then(function() {
```

Comment:
This .then/.catch pattern looks like it really doesn't do anything. Is it just to show how you could handle an error if you wanted to?

Reply:
It's to show that the developer is able to run more code after showing the task pane. For e.g. a way this can be used is to show the taskpane and then update the value in the taskpane with an animation.